contracts/wallet/WalletFactory.sol

```diff
@@ -44,95 +44,49 @@ contract WalletFactory is Owned, Managed {
 
     event ModuleRegistryChanged(address addr);
     event ENSManagerChanged(address addr);
-    event GuardianStorageChanged(address addr);
     event WalletCreated(address indexed wallet, address indexed owner, address indexed guardian);
 
-    // *************** Modifiers *************************** //
-
-    /**
-     * @dev Throws if the guardian storage address is not set.
-     */
-    modifier guardianStorageDefined {
-        require(guardianStorage != address(0), "IGuardianStorage address not defined");
-        _;
-    }
-
     // *************** Constructor ********************** //
 
     /**
      * @dev Default constructor.
      */
-    constructor(address _moduleRegistry, address _walletImplementation, address _ensManager) public {
+    constructor(address _moduleRegistry, address _walletImplementation, address _ensManager, address _guardianStorage) public {
```

Comment:
Now I remember why the guardianStorage was provided with a setter: it is because in the deployment scripts we deploy the wallet factory before we deploy the modules and their storage. If the guardian storage is passed in the constructor of the factory we will need to update the scripts and the order in which contracts are deployed.

Reply:
That's fine I think. Normally I'd expect storage contracts to be deployed first so I think I prefer that option. 
I've fixed the deployment scripts which I hadn't updated with the new constructor though when I was looking at this.